source/extensions/quic_listeners/quiche/active_quic_listener.cc

```diff
@@ -0,0 +1,76 @@
+#include "extensions/quic_listeners/quiche/active_quic_listener.h"
+
+#include "extensions/quic_listeners/quiche/envoy_quic_alarm_factory.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_connection_helper.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_dispatcher.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_fake_proof_source.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_packet_writer.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_utils.h"
+
+namespace Envoy {
+namespace Quic {
+
+ActiveQuicListener::ActiveQuicListener(Event::Dispatcher& dispatcher,
+                                       Network::ConnectionHandler& parent, spdlog::logger& logger,
+                                       Network::ListenerConfig& listener_config,
+                                       const quic::QuicConfig& quic_config)
+    : ActiveQuicListener(dispatcher, parent,
+                         dispatcher.createUdpListener(listener_config.socket(), *this), logger,
+                         listener_config, quic_config) {}
+
+ActiveQuicListener::ActiveQuicListener(Event::Dispatcher& dispatcher,
+                                       Network::ConnectionHandler& parent,
+                                       Network::ListenerPtr&& listener, spdlog::logger& logger,
+                                       Network::ListenerConfig& listener_config,
+                                       const quic::QuicConfig& quic_config)
+    : Server::ConnectionHandlerImpl::ActiveListenerImplBase(std::move(listener), listener_config),
+      logger_(logger), dispatcher_(dispatcher), version_manager_(quic::CurrentSupportedVersions()) {
+  quic::QuicRandom* const random = quic::QuicRandom::GetInstance();
+  random->RandBytes(random_seed_, sizeof(random_seed_));
+  crypto_config_ = std::make_unique<quic::QuicCryptoServerConfig>(
+      quic::QuicStringPiece(reinterpret_cast<char*>(random_seed_), sizeof(random_seed_)),
+      quic::QuicRandom::GetInstance(), std::make_unique<EnvoyQuicFakeProofSource>(),
+      quic::KeyExchangeSource::Default());
+  auto connection_helper = std::make_unique<EnvoyQuicConnectionHelper>(dispatcher_);
+  auto alarm_factory =
+      std::make_unique<EnvoyQuicAlarmFactory>(dispatcher_, *connection_helper->GetClock());
+  quic_dispatcher_ = std::make_unique<EnvoyQuicDispatcher>(
+      crypto_config_.get(), quic_config, &version_manager_, std::move(connection_helper),
+      std::move(alarm_factory), quic::kQuicDefaultConnectionIdLength, parent, config_, stats_,
+      dispatcher);
+  quic_dispatcher_->InitializeWithWriter(
+      new EnvoyQuicPacketWriter(*dynamic_cast<Network::UdpListener*>(listener_.get())));
```

Comment:
Can you add a TODO here to try to get rid of the dynamic cast? I will try to look at this myself and provide some guidance instead of just complaining. :)